src/ray/core_worker/core_worker_test.cc

```diff
@@ -1,20 +1,130 @@
+#include <thread>
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
 
 #include "core_worker.h"
+#include "context.h"
 #include "ray/common/buffer.h"
+#include "ray/raylet/raylet_client.h"
+
+#include <boost/asio.hpp>
+#include <boost/asio/error.hpp>
+#include <boost/bind.hpp>
+
+#include "ray/thirdparty/hiredis/async.h"
+#include "ray/thirdparty/hiredis/hiredis.h"
 
 namespace ray {
 
+std::string store_executable;
+std::string raylet_executable;
+
+ray::ObjectID RandomObjectID() { return ObjectID::from_random(); }
+
+static void flushall_redis(void) {
+  redisContext *context = redisConnect("127.0.0.1", 6379);
+  freeReplyObject(redisCommand(context, "FLUSHALL"));
+  freeReplyObject(redisCommand(context, "SET NumRedisShards 1"));
+  freeReplyObject(redisCommand(context, "LPUSH RedisShards 127.0.0.1:6380"));  
+  redisFree(context);
+}
+
 class CoreWorkerTest : public ::testing::Test {
  public:
-  CoreWorkerTest() : core_worker_(WorkerType::WORKER, Language::PYTHON) {}
+  CoreWorkerTest(int num_nodes)
+    : raylet_socket_names_(num_nodes),
+      raylet_store_socket_names_(num_nodes) {
+    RAY_CHECK(num_nodes > 0);
+    // start plasma store.
+    for (auto &store_socket : raylet_store_socket_names_) {
+      store_socket = StartStore();
+    }
+
+    // start raylet on head node
+    for (int i = 0; i < num_nodes; i++) {
+      raylet_socket_names_[i] = StartRaylet(raylet_store_socket_names_[i],
+         "127.0.0.1", "127.0.0.1", "\"CPU,4.0,resA,100\"");
```

Comment:
```suggestion
         "127.0.0.1", "127.0.0.1", "\"CPU,4.0\"");
```
`resA` isn't used, let's add this later if needed.